components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -3308,6 +3308,23 @@ public String getOpenAPIDefinition(String apiId, String organization) throws API
         return APIUtil.removeXMediationScriptsFromSwagger(definition);
     }
 
+    @Override
+    public String invokeApiChatExecute(String apiChatRequestId, String requestPayload) throws APIManagementException {
+        return APIUtil.invokeAIService(APIConstants.API_CHAT_ENDPOINT,
+                APIConstants.API_CHAT_AUTH_TOKEN, APIConstants.API_CHAT_EXECUTE_RESOURCE, requestPayload,
+                apiChatRequestId);
+    }
+
+    @Override
+    public String invokeApiChatPrepare(String apiId, String apiChatRequestId, String organization)
+            throws APIManagementException {
+        String swaggerDefinition = getOpenAPIDefinition(apiId, organization);
+        String payload = "{\"openapi\": " + swaggerDefinition + "}";
```

Comment:
Shall remove the use of magic strings?

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12374/files#diff-eea96b2f615001930c3a7cf2ec3802f9dc86ac99cabcd2ae10716b934c76c3a2L3328-R3340